src/cache.rs

```diff
@@ -659,14 +659,18 @@ impl Cache {
     /// corresponding source has been parsed.
     /// If resolved imports contain imports themselves, resolve them recursively.
     /// It returns a `Vec<FileId>`, which is a vector of all the imports that were transitively resolved.
+    #[allow(clippy::type_complexity)]
     pub fn resolve_imports(
         &mut self,
         file_id: FileId,
-    ) -> Result<CacheOp<Vec<FileId>>, CacheError<ImportError>> {
+        resillient: bool,
```

Comment:
Please, no :cry:  we've had a lot of bugs (and still have, in fact) because of those pesky error-tolerance booleans that were present in each and every function of the cache, and it's really hard to keep track on what should be error tolerant or not. Now, there is a data `error_tolerance` or something in the cache that dictates the behavior.  There is an `ErrorTolerance` type. You should probably use it instead and not take any additional argument

Reply:
Right. This is much better. Thank you! 🙂